File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbItemCat;
import com.pinyougou.utils.PageResoult;
import com.pinyougou.utils.Resoult;

public interface ItemCatService {
    PageResoult<TbItemCat> findAll(long grade);

    PageResoult<TbItemCat> searchByCondition(int currentPage, int pageSize, TbItemCat itemCat);

    Resoult addItemCat(TbItemCat itemCat);

    Resoult updateItemCat(TbItemCat itemCat);

    Resoult deleteItemCat(String[] selectedItems);

    PageResoult<TbItemCat> findItemCatByParentId(long parentId);

    TbItemCat findOne(long id);

    PageResoult<TbItemCat> findAllItemCat();
}

File: pinyougou-shop-web/src/main/webapp/service/itemCatservice.js
app.service("itemCatService",function ($http) {

    //查询全部分类
    this.findAll = function (grade) {
        return $http.get("/itemCat/list.do?grade=" + grade);
    }
    //条件查询分类带分页
    this.searchByCondition = function (currentPage,itemsPerPage,seachEntity) {
        return $http.post("/itemCat/searchByCondition.do?currentPage=" + currentPage + "&pageSize=" + itemsPerPage,seachEntity);
    }

    this.save=function (url,itemCat) {
        return $http.post(url,itemCat);
    }
    this.dele=function(selectItems){
        return $http.post("/itemCat/deleteItemCat.do",selectItems);
    }
    this.findItemCatByParentId=function (parentId) {
        return $http.get("/itemCat/findItemCatByParentId.do?parentId=" + parentId);
    }
    this.findOne=function (id) {
        return $http.get("/itemCat/findOne.do?id=" +id);
    }
    this.findAllItemCat=function () {
        return $http.get("/itemCat/findAllItemCat.do");
    }

})